lib/converter.js

```diff
@@ -44,15 +45,16 @@ async function convert(inputFile, options, timings) {
     doc = asciidoctor.loadFile(inputFile, options)
     html = doc.convert(options)
   }
-  const workingDir = path.dirname(inputFile)
-  const inputFilenameWithoutExt = path.basename(inputFile, path.extname(inputFile))
-  const outputFile = path.join(workingDir, inputFilenameWithoutExt + '.pdf')
-  let tempFile;
-  if (path.isAbsolute(workingDir)) {
-    tempFile = path.join(workingDir, inputFilenameWithoutExt + '_temp.html')
-  } else {
-    tempFile = path.normalize(path.join(process.cwd(), workingDir, inputFilenameWithoutExt + '_temp.html'))
+
+  let toDir = options.to_dir || path.dirname(inputFile)
+  if (!path.isAbsolute(toDir)) {
+    toDir = path.normalize(path.join(process.cwd(), toDir))
   }
+  await ensureExists(toDir)
```

Comment:
I'm a bit confused, we should ensure that the directory exists before the conversion no ?

Reply:
Yes, and this is exactly what is being done here. That is, If I remove this line, `writeFile` bellow will fail because the directory does not exist.

Reply:
Ah, I think I get while we are confused!

We use ` html = doc.convert(options)` which I think is a pure function which looks like a pure function which does not write any output files. We do writing of the files ourselves, and that causes problems.

I think if we just use `asciidoctor.convertFile`, we will be OK?

We'll need to setup options in a special way to get the `_temp.html` suffix though, but perhaps the `_temp` is not really neccessary?

Reply:
Hm, so the problem with `convertFile` is that there isn't an easy way to get the path of the result.

I see two options here:

* we use `convertFile` with `to_file: true` and try to infer the correct output file from `options` ourselves. That will match upstream cli, but will result in a bug if own logic and upstream logic diverges. 

* we continue using `convertFile` with `to_file: false`, and write our own logic to determine output dir, etc. This will work correctly, but may diverge from upstream in behavior.

Reply:
I actually wonder if it makes sense to just upstream this to asciidoctor-js?

The core feature here, as I understand it, is the ability to define your own HTML templates using JavaScript. If `asciidoctor-js` had it out of the box, conversion to PDF could be achieved with a simple bash script.